client/state/document-head/test/reducer.js

```diff
@@ -32,13 +32,6 @@ describe( 'reducer', () => {
 
 			expect( newState ).to.equal( 'new title' );
 		} );
-
-		it( 'should return initial state on route set action', () => {
```

Comment:
Would it make sense to keep this and update the check to `should preserve existing title on route set action`?